src/core/buffer/representation_buffer.ts

```diff
@@ -183,10 +185,13 @@ export default function RepresentationBuffer<T>({
   // SourceBuffer.
   const sourceBufferWaitingQueue = new SimpleSet();
 
+  const finishedQueue$ = new Subject<void>();
+
   const status$ = observableCombineLatest(
     clock$,
     wantedBufferAhead$,
-    terminate$.pipe(mapTo(true), startWith(false))
+    terminate$.pipe(take(1), mapTo(true), startWith(false)),
+    finishedQueue$.pipe(startWith(undefined))
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this redundant "undefined". ([typescript:S4623](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4623))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=354&issues=AWaq2qEPFsWw174ZFNPE&open=AWaq2qEPFsWw174ZFNPE)

[ISSUE_KEY]: AWaq2qEPFsWw174ZFNPE